docs/modules/migrate/pages/upgrading-from-jet.adoc

```diff
@@ -0,0 +1,197 @@
+
+=== Configuration Files
+
+The Jet configuration elements have been added to
+the XML/YAML files of IMDG. For the Jet engine related configuration elements, see the
+https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/resources/hazelcast-full-example.yaml#L3490[example YAML file on GitHub].
+
+Hazelcast 5.0 checks and validates your YAML configurations during a cluster startup.
+According to this validation:
+
+* The top-level `hazelcast` object must exist. 
+* Client and member YAML configurations must be separate (not in the same file).
+* There must be no case insensitive enum values.
+
+While upgrading to Hazelcast 5.0, if a YAML configuration violates any of these rules,
+the cluster will not start. You need to either edit and update your YAML configuration files
+accordingly or disable the validation by setting the `hazelcast.config.schema.validation.enabled` property to `false`.
+
+== Entry Points
+
+The `Jet` class, which was the main entry point of Jet 4.x,
+has been deprecated and replaced by the link:https://docs.hazelcast.org/docs/{page-component-version}/javadoc/com/hazelcast/core/HazelcastInstance.html[`HazelcastInstance` class].
+
+The `JetInstance` class, which
+represented an instance of a Jet member or client has been been deprecated and replaced by the link:https://docs.hazelcast.org/docs/{page-component-version}/javadoc/com/hazelcast/jet/JetService.html[`JetService` class].
+
+The`JetInstance.bootstrappedInstance()` method been deprecated and replaced by `HazelcastInstance.bootstrappedInstance()`.
+
+Use the following table to find out which new class to use instead of JetInstance`.
+
+.JetInstance Replacements
+[cols="1a,1m"]
+|===
+|JetInstance Usage|New class
+
+| Submitting streaming/batch jobs to the cluster and managing them.
+|JetService
+
+|Accessing Hazelcast data structures.
+a|`HazelcastInstance`
+
+The only exception is Jet observables. An observable is a Jet data structure and we ported it to the `JetService` class. 
+
+|Performing cluster operations such as shutting down the cluster.
+|HazelcastInstance
+|===
+
+To access Jet related services, you should now use the link:https://docs.hazelcast.org/docs/{page-component-version}/javadoc/com/hazelcast/core/HazelcastInstance.html#getJet--[`HazelcastInstance.getJet()` method] to get an instance of the `JetService` object.
+
+[tabs] 
+==== 
+Jet 4.x:: 
++ 
+-- 
+```java
+HazelcastInstance hz = Hazelcast.newHazelcastInstance();
+JetInstance jet = hz.getJetInstance();
+```
+--
+Hazelcast:: 
++ 
+-- 
+```java
+HazelcastInstance hz = Hazelcast.newHazelcastInstance();
+JetService jet = hz.getJet();
+```
+--
+====
+
+== Enabling the Jet Engine
+
+In embedded mode (Hazelcast as a classic Java process), Jet is disabled by default. As a result, you cannot use Jet APIs or SQL unless you enable the Jet engine.
```

Comment:
Hazelcast always runs as a Java process. The embedded mode refers to "Hazelcast running in the JVMs of your application" as opposed to "Hazelcast running in its own standalone JVMs.